whois-endtoend/src/test/groovy/net/ripe/db/whois/spec/update/RoleSpec.groovy

```diff
@@ -709,6 +709,48 @@ class RoleSpec extends BaseQueryUpdateSpec {
         query_object_matches("-T role FR1-TEST", "role", "Abuse Role", "email@xn--zrich-kva.example")
     }
 
+    def "Abuse-mailbox with Umlaut IDN Converted to Punycode using HTTP"() {
+        given:
+
+        expect:
+        queryObjectNotFound("-r -T role FR1-TEST", "role", "Abuse Role")
+
+        when:
+        def message = syncUpdateHttp("""
+                role:          Abuse Role
+                address:       St James Street
+                address:       Burnley
+                address:       UK
+                e-mail:        dbtest@ripe.net
+                abuse-mailbox: email@zürich.example
+                nic-hdl:       FR1-TEST
+                mnt-by:        owner-mnt
+                source:        TEST
+
+                password: owner
+                """.stripIndent(true)
+        )
+
+        then:
+        def ack = new AckResponse("", message)
+
+        ack.summary.nrFound == 1
+        ack.summary.assertSuccess(1, 1, 0, 0, 0)
+        ack.summary.assertErrors(0, 0, 0, 0)
+
+        ack.countErrorWarnInfo(0, 3, 0)
+        ack.successes.any { it.operation == "Create" && it.key == "[role] FR1-TEST   Abuse Role" }
+        def warnings = ack.allWarnings
+        warnings[0] == "This Syncupdates request used insecure HTTP, which may be removed in\n" +
+                "            a future release. Please switch to HTTPS."
```

Comment:
There is a method to assert with order warningSuccessMessagesFor. It also takes care of this assert  ack.successes.any { it.operation == "Create" && it.key == "[role] FR1-TEST   Abuse Role" }

Reply:
doesn't take the global warnings, just the specific warnings